apps/allocations/app/components/Panel/NewBudget.js

```diff
@@ -0,0 +1,99 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { DropDown, TextInput } from '@aragon/ui'
+import styled from 'styled-components'
+
+import { Form, FormField } from '../Form'
+import { isStringEmpty } from '../../utils/helpers'
+import { MIN_AMOUNT } from '../../utils/constants'
+
+// TODO:: This should be votingTokens from account?
+const INITIAL_STATE = {
+  name: '',
+  nameError: true,
+  amount: '',
+  amountError: true,
+  currency: 0,
+}
+
+class NewBudget extends React.Component {
+  static propTypes = {
+    onCreateBudget: PropTypes.func.isRequired
+  }
+
+  state =  INITIAL_STATE
+
+  changeField = e => {
+    const { name, value } = e.target
+    this.setState({ [e.target.name]: e.target.value })
```

Comment:
Since we are destructuring the `e.target` object we don't need to call it again:
```suggestion
    this.setState({ [name]: value })
```
(Otherwise keep this line and remove the previous one, whatever makes you happier 👍